diff --git a/packages/ui/src/app/views/modals/actions-modal/actions-modal.tsx b/packages/ui/src/app/views/modals/actions-modal/actions-modal.tsx
index 99ad24fd..5cde0dc2 100644
--- a/packages/ui/src/app/views/modals/actions-modal/actions-modal.tsx
+++ b/packages/ui/src/app/views/modals/actions-modal/actions-modal.tsx
@@ -13,6 +13,7 @@ export const ActionsModal: Component = () => {
             enableAndroidBackHandler={appState.enableAndroidBackHandler}
             onClose={() => setAction(null)}
             data-tc-actions-modal-container="true"
+            class={`${action()!.name}-modal`}
         >
             <Switch>
                 <Match when={action()!.name === 'transaction-sent'}>
diff --git a/packages/ui/src/app/views/modals/wallets-modal/all-wallets-list-modal/index.tsx b/packages/ui/src/app/views/modals/wallets-modal/all-wallets-list-modal/index.tsx
index 2e49c3b7..b5506470 100644
--- a/packages/ui/src/app/views/modals/wallets-modal/all-wallets-list-modal/index.tsx
+++ b/packages/ui/src/app/views/modals/wallets-modal/all-wallets-list-modal/index.tsx
@@ -59,7 +59,7 @@ export const AllWalletsListModal: Component<DesktopSelectWalletModalProps> = pro
         walletsList().filter(wallet => !wallet.isSupportRequiredFeatures);

     return (
-        <DesktopSelectWalletModalStyled data-tc-wallets-modal-list="true">
+        <DesktopSelectWalletModalStyled data-tc-wallets-modal-list="true" class="all-wallets-list-modal">
             <StyledIconButton icon="arrow" onClick={() => props.onBack()} />
             <H1Styled translationKey="walletModal.wallets">Wallets</H1Styled>
             <ScrollContainer maxHeight={maxHeight()}>
diff --git a/packages/ui/src/app/views/modals/wallets-modal/desktop-connection-modal/index.tsx b/packages/ui/src/app/views/modals/wallets-modal/desktop-connection-modal/index.tsx
index 19c7ed31..77fb9458 100644
--- a/packages/ui/src/app/views/modals/wallets-modal/desktop-connection-modal/index.tsx
+++ b/packages/ui/src/app/views/modals/wallets-modal/desktop-connection-modal/index.tsx
@@ -200,7 +200,7 @@ export const DesktopConnectionModal: Component<DesktopConnectionProps> = props =
     }

     return (
-        <DesktopConnectionModalStyled data-tc-wallets-modal-connection-desktop="true">
+        <DesktopConnectionModalStyled data-tc-wallets-modal-connection-desktop="true" class="desktop-connection-modal">
             <Show when={!props.backDisabled}>
                 <StyledIconButton icon="arrow" onClick={() => props.onBackClick()} />
             </Show>
diff --git a/packages/ui/src/app/views/modals/wallets-modal/desktop-universal-modal/index.tsx b/packages/ui/src/app/views/modals/wallets-modal/desktop-universal-modal/index.tsx
index af6b9b26..b1b170eb 100644
--- a/packages/ui/src/app/views/modals/wallets-modal/desktop-universal-modal/index.tsx
+++ b/packages/ui/src/app/views/modals/wallets-modal/desktop-universal-modal/index.tsx
@@ -56,6 +56,7 @@ export const DesktopUniversalModal: Component<DesktopUniversalModalProps> = prop
         <DesktopUniversalModalStyled
             onClick={() => setPopupOpened(false)}
             data-tc-wallets-modal-universal-desktop="true"
+            class="desktop-universal-modal"
         >
             <H1 translationKey="walletModal.desktopUniversalModal.connectYourWallet">
                 Connect your wallet
diff --git a/packages/ui/src/app/views/modals/wallets-modal/info-modal/index.tsx b/packages/ui/src/app/views/modals/wallets-modal/info-modal/index.tsx
index 4d4c597c..65b64e65 100644
--- a/packages/ui/src/app/views/modals/wallets-modal/info-modal/index.tsx
+++ b/packages/ui/src/app/views/modals/wallets-modal/info-modal/index.tsx
@@ -27,7 +27,7 @@ interface InfoModalProps {

 export const InfoModal: Component<InfoModalProps> = props => {
     return (
-        <InfoModalStyled data-tc-wallets-modal-info="true">
+        <InfoModalStyled data-tc-wallets-modal-info="true" class="info-modal">
             <StyledIconButton icon="arrow" onClick={() => props.onBackClick()} />
             <H1Styled translationKey="walletModal.infoModal.whatIsAWallet">
                 What is a wallet
diff --git a/packages/ui/src/app/views/modals/wallets-modal/mobile-connection-modal/index.tsx b/packages/ui/src/app/views/modals/wallets-modal/mobile-connection-modal/index.tsx
index ecfdebb8..4624059b 100644
--- a/packages/ui/src/app/views/modals/wallets-modal/mobile-connection-modal/index.tsx
+++ b/packages/ui/src/app/views/modals/wallets-modal/mobile-connection-modal/index.tsx
@@ -154,7 +154,7 @@ export const MobileConnectionModal: Component<MobileConnectionProps> = props =>
     onRetry();

     return (
-        <MobileConnectionModalStyled data-tc-wallets-modal-connection-mobile="true">
+        <MobileConnectionModalStyled data-tc-wallets-modal-connection-mobile="true" class="mobile-connection-modal">
             <Show when={!props.backDisabled || showQR()}>
                 <StyledIconButton icon="arrow" onClick={onBack} />
             </Show>
diff --git a/packages/ui/src/app/views/modals/wallets-modal/mobile-universal-modal/index.tsx b/packages/ui/src/app/views/modals/wallets-modal/mobile-universal-modal/index.tsx
index 55d48e61..e201894e 100644
--- a/packages/ui/src/app/views/modals/wallets-modal/mobile-universal-modal/index.tsx
+++ b/packages/ui/src/app/views/modals/wallets-modal/mobile-universal-modal/index.tsx
@@ -150,7 +150,7 @@ export const MobileUniversalModal: Component<MobileUniversalModalProps> = props
     );

     return (
-        <div data-tc-wallets-modal-universal-mobile="true">
+        <div data-tc-wallets-modal-universal-mobile="true" class="mobile-universal-modal">
             <Show when={showQR()}>
                 <StyledLeftActionButton icon="arrow" onClick={onCloseQR} />
                 <MobileUniversalQR
diff --git a/packages/ui/src/app/views/modals/wallets-modal/single-wallet-modal.tsx b/packages/ui/src/app/views/modals/wallets-modal/single-wallet-modal.tsx
index 53df744a..9399d866 100644
--- a/packages/ui/src/app/views/modals/wallets-modal/single-wallet-modal.tsx
+++ b/packages/ui/src/app/views/modals/wallets-modal/single-wallet-modal.tsx
@@ -75,6 +75,7 @@ export const SingleWalletModal: Component = () => {
             onClose={() => onClose('action-cancelled')}
             onClickQuestion={() => setInfoTab(v => !v)}
             data-tc-wallets-modal-container="true"
+            class="single-wallet-modal"
         >
             <Show when={infoTab()}>
                 <InfoModal onBackClick={() => setInfoTab(false)} />
diff --git a/packages/ui/src/app/views/modals/wallets-modal/wallets-modal.tsx b/packages/ui/src/app/views/modals/wallets-modal/wallets-modal.tsx
index d370a41e..5f768858 100644
--- a/packages/ui/src/app/views/modals/wallets-modal/wallets-modal.tsx
+++ b/packages/ui/src/app/views/modals/wallets-modal/wallets-modal.tsx
@@ -182,6 +182,7 @@ export const WalletsModal: Component = () => {
             onClose={() => onClose('action-cancelled')}
             onClickQuestion={() => setInfoTab(v => !v)}
             data-tc-wallets-modal-container="true"
+            class="wallets-modal"
         >
             <Show when={infoTab()}>
                 <InfoModal onBackClick={() => setInfoTab(false)} />
